### PR TITLE
Optimize listing of project files

### PR DESCRIPTION
**Pull request by nglgzz**
### Please check if the PR fulfills these requirements
- [x] The commit message follows our [guidelines](https://github.com/neo4j-devtools/relate/blob/master/CONTRIBUTING.md)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

### What kind of change does this PR introduce?
Performance improvement

### What is the current behavior?
- Adding a big directory (eg. the relate repository containing over 200k files) as a project and then listing the files in that project makes the process run out of memory and crash. Filtering out the directories containing most files (ie. node_modules) has no effect on the outcome.
- If there is a directory the user has no read access to inside the project, listing files will fail.

### What is the new behavior?
- The process no longer crashes with big directories, and when filtering out directories, those directories are not read at all.
- If the user has no access to a directory inside their project, that directory is ignored.

### Does this PR introduce a breaking change?
No

### CODE REVIEW

File: packages/cli/src/modules/project/list-files.module.ts [hunk 1]
```diff
@@ -19,13 +19,43 @@ export class ListFilesModule implements OnApplicationBootstrap {
...
         const environment = await this.systemProvider.getEnvironment(environmentId);
         const projectId = flags.project || (await selectProjectPrompt('Select a Project', environment));
-        const files = await environment.projects.listFiles(projectId);
+
+        let {ignore = []} = flags;
+        // This is to accept both common formats for arrays in the CLI:
+        // command -i val1 -i val2 -i val3
+        // command -i val1,val2,val3
+        if (ignore.length === 1) {
+            ignore = ignore[0].split(',');
```
Comment (jk05) [inline packages/cli/src/modules/project/list-files.module.ts:31]: 👍

File: packages/cli/src/modules/project/list-files.module.ts [hunk 2]
```diff
@@ -19,13 +19,43 @@ export class ListFilesModule implements OnApplicationBootstrap {
...
+                      field: 'directory',
+                      type: FILTER_COMPARATORS.NOT_CONTAINS,
+                      value: 'node_modules',
+                  },
+                  {
+                      field: 'directory',
+                      type: FILTER_COMPARATORS.NOT_CONTAINS,
+                      value: '.git',
+                  },
+              ];
```
Comment (jk05) [inline packages/cli/src/modules/project/list-files.module.ts:52]: should we do this within Relate itself?

File: packages/cli/docs/project.md [hunk 3]
```diff
@@ -178,7 +178,16 @@ USAGE
...
+  -x, --extended                 show extra columns
+  --columns=columns              only show provided columns (comma-separated)
+  --filter=filter                filter property by partial string matching, ex: name=foo
+  --no-header                    hide table header from output
+  --no-truncate                  do not truncate output to fit screen
+  --output=csv|json|yaml         output in a more machine friendly format
+  --sort=sort                    property to sort by (prepend '-' for descending)
 
 EXAMPLES
   $ relate project:list-files
```
Comment (jk05) [inline packages/cli/docs/project.md:193]: add an example in particular for `ignore`?

File: packages/cli/src/modules/project/list-files.module.ts (hunk 2 above)
Comment (nglgzz) [reply packages/cli/src/modules/project/list-files.module.ts:52]: I don't think we should add this to the common package, but it feels like a sensible default for the CLI. I will change this to be a default value for the flag though, in the current format there's no way for the user to know which folders we're ignoring by default.

Comment (jk05) [reply packages/cli/src/modules/project/list-files.module.ts]: right in terms of the CLI output, Im with you.

**Review by jk05 — APPROVED** [review]

**Pull request merged**